File: src/main/java/dev/robothanzo/werewolf/database/documents/AuthSession.java
package dev.robothanzo.werewolf.database.documents;

import com.mongodb.client.MongoCollection;
import dev.robothanzo.werewolf.database.Database;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.concurrent.TimeUnit;
import com.mongodb.client.model.IndexOptions;
import com.mongodb.client.model.Indexes;

import static com.mongodb.client.model.Filters.eq;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthSession {
    private String sessionId;
    private String userId;
    private String username;
    private String discriminator;
    private String avatar;
    private long guildId;
    private String role;
    private Date createdAt;

    public static MongoCollection<AuthSession> fetchCollection() {
        MongoCollection<AuthSession> collection = Database.database.getCollection("auth_sessions", AuthSession.class);
        // Create TTL index on createdAt if it doesn't exist (30 days)
        collection.createIndex(Indexes.ascending("createdAt"), new IndexOptions().expireAfter(30L, TimeUnit.DAYS));
        return collection;
    }
}

File: src/main/kotlin/dev/robothanzo/werewolf/database/documents/AuthSession.kt
package dev.robothanzo.werewolf.database.documents

import org.bson.codecs.pojo.annotations.BsonIgnore
import java.io.Serializable
import java.util.*

data class AuthSession(
    var sessionId: String? = null,
    var userId: String? = null,
    var guildId: String? = null,
    var role: UserRole? = null,
    var createdAt: Date? = null
) : Serializable {

    @get:BsonIgnore
    val isJudge: Boolean
        get() = role == UserRole.JUDGE

    @get:BsonIgnore
    val isSpectator: Boolean
        get() = role == UserRole.SPECTATOR

    @get:BsonIgnore
    val isPrivileged: Boolean
        get() = role != null && role!!.isPrivileged()

    @get:BsonIgnore
    val isBlocked: Boolean
        get() = role == UserRole.BLOCKED

    @get:BsonIgnore
    val isPending: Boolean
        get() = role == UserRole.PENDING || role == null
}

enum class UserRole {
    JUDGE,
    SPECTATOR,
    PENDING,
    BLOCKED;

    fun isPrivileged(): Boolean {
        return this == JUDGE || this == SPECTATOR
    }

    companion object {
        fun fromString(role: String?): UserRole {
            if (role == null) return PENDING
            return try {
                valueOf(role.uppercase())
            } catch (e: IllegalArgumentException) {
                PENDING
            }
        }
    }
}